presto-bytecode/src/main/java/com/facebook/presto/bytecode/StaticTypeBytecodeExpression.java

```diff
@@ -92,4 +93,21 @@ public BytecodeExpression writeValue(BytecodeExpression blockBuilder, BytecodeEx
         }
         return invoke("writeObject", void.class, blockBuilder, value.cast(Object.class));
     }
+
+    public BytecodeExpression writeValueWithCast(BytecodeExpression blockBuilder, BytecodeExpression value)
```

Comment:
Maybe good to add a javadoc to explain this might be used by plugins? (Because it is not used anywhere in prestodb).